File: test-suite-java/src/test/java/io/micronaut/interceptor/test/micronautapi/CompiledService.java
package io.micronaut.interceptor.test.micronautapi;

import io.micronaut.context.annotation.Parameter;
import io.micronaut.context.annotation.Prototype;
import jakarta.annotation.PostConstruct;

@Prototype
@Compiled(region = "users")
public class CompiledService {

    private final String name;

    public CompiledService(@Parameter String name) {
        this.name = name;
    }

    @PostConstruct
    void started() {
    }

    public String greet(String greeting) {
        return greeting + " " + name;
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/interceptor/test/CompiledService.kt
package io.micronaut.interceptor.test

import io.micronaut.context.annotation.Parameter
import io.micronaut.context.annotation.Prototype
import jakarta.annotation.PostConstruct

@Prototype
@Compiled(region = "users")
open class CompiledService(@param:Parameter private val name: String) {

    @PostConstruct
    open fun started() {
    }

    open fun greet(greeting: String): String = "$greeting $name"
}
